js/__internal/grids/grid_core/keyboard_navigation/module_utils.ts

```diff
@@ -0,0 +1,54 @@
+import { isDefined } from '@js/core/utils/type';
+import devices from '@js/core/devices';
+import { EDITOR_CELL_CLASS } from '@js/ui/grid_core/ui.grid_core.editing_constants';
+import {
+  GROUP_ROW_CLASS, MASTER_DETAIL_ROW_CLASS, FREESPACE_ROW_CLASS, VIRTUAL_ROW_CLASS, COMMAND_SELECT_CLASS, HEADER_ROW_CLASS,
+} from './const';
+
+export function isGroupRow($row) {
+  return $row && $row.hasClass(GROUP_ROW_CLASS);
+}
+
+export function isDetailRow($row) {
+  return $row && $row.hasClass(MASTER_DETAIL_ROW_CLASS);
+}
+
+export function isDataRow($row) {
+  return $row && !isGroupRow($row) && !isDetailRow($row);
+}
+
+export function isNotFocusedRow($row) {
+  return !$row || ($row.hasClass(FREESPACE_ROW_CLASS) || $row.hasClass(VIRTUAL_ROW_CLASS));
```

Comment:
Let's remove redundant brackets:

```suggestion
  return !$row || $row.hasClass(FREESPACE_ROW_CLASS) || $row.hasClass(VIRTUAL_ROW_CLASS);
```